hosted/src/spi.rs

```diff
@@ -73,7 +73,7 @@ impl<V: Version<BusItem = u8>> Spi<V> {
     {
         let target: Vec<u8> = Message::<V>::new(payload).collect();
         // Make a really large buffer, this is a "fulhack" but it is fine for now.
-        let mut rx_buff = [0; 1024];
+        let mut rx_buff: Vec<u8>  = target.iter().map(|_| 0).collect();
```

Comment:
This should be fixed in the future as this `might` cause issues with future versions of the protocol. the current versions always allocate the maximum space needed.